ui/src/app/subnets/views/VLANDetails/VLANDetailsHeader/VLANDeleteForm/VLANDeleteForm.tsx

```diff
@@ -0,0 +1,61 @@
+import { useCallback } from "react";
+
+import { Icon } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import FormikForm from "app/base/components/FormikForm";
+import type { EmptyObject } from "app/base/types";
+import type { RootState } from "app/store/root/types";
+import { actions as vlanActions } from "app/store/vlan";
+import vlanSelectors from "app/store/vlan/selectors";
+import type { VLAN, VLANMeta } from "app/store/vlan/types";
+import subnetURLs from "app/subnets/urls";
+
+type Props = {
+  closeForm: () => void;
+  id?: VLAN[VLANMeta.PK] | null;
+};
+
+const VLANDeleteForm = ({ closeForm, id }: Props): JSX.Element | null => {
+  const dispatch = useDispatch();
+  const vlan = useSelector((state: RootState) =>
+    vlanSelectors.getById(state, id)
+  );
+  const errors = useSelector(vlanSelectors.errors);
+  const saved = useSelector(vlanSelectors.saved);
+  const saving = useSelector(vlanSelectors.saving);
+  const cleanup = useCallback(() => vlanActions.cleanup(), []);
+
+  if (!id || !vlan) {
```

Comment:
Should we use the `isId()` util? Or really just checking the existence of `vlan` is enough here.

Reply:
Yep you're totes right, I'll fix it up.